kratos/utilities/geometry_utilities/brep_sbm_utilities.h

```diff
@@ -0,0 +1,106 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+
+#pragma once
+
+// Std includes
+#include <list>
+
+// System includes
+#include "includes/define.h"
+
+// Project includes
+#include "geometries/geometry.h"
+#include "geometries/brep_curve_on_surface.h"
+
+#include "includes/node.h"
+
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+template<bool TShiftedBoundary, class TNodeType = Node>
+class KRATOS_API(KRATOS_CORE) BrepSbmUtilities
+{
+public:
+
+    KRATOS_CLASS_POINTER_DEFINITION(BrepSbmUtilities);
+    ///@name Type Definitions
+    ///@{
+
+    typedef std::size_t IndexType;
+    typedef std::size_t SizeType;
+
+    typedef IntegrationPoint<3> IntegrationPointType;
+    typedef std::vector<IntegrationPointType> IntegrationPointsArrayType;
+
+    using BrepCurveOnSurfacePointerType = typename BrepCurveOnSurface<PointerVector<Node>, TShiftedBoundary, PointerVector<Point>>::Pointer;
+
+    using GeometryType = Geometry<TNodeType>;
+    using GeometryPointerType = typename GeometryType::Pointer;
+    using GeometrySurrogateArrayType = DenseVector<GeometryPointerType>;
+
+    /**
+     * @brief Create a Brep Surface Sbm Integration Points object
+     * 
+     * @param rSpansU 
+     * @param rSpansV 
+     * @param rSurrogateOuterLoopGeometries 
+     * @param rSurrogateInnerLoopGeometries 
+     * @param rIntegrationPoints 
+     * @param rIntegrationInfo 
+     */
+    static void CreateBrepSurfaceSbmIntegrationPoints(
+        const std::vector<double>& rSpansU,
+        const std::vector<double>& rSpansV,
+        const GeometrySurrogateArrayType& rSurrogateOuterLoopGeometries,
+        const GeometrySurrogateArrayType& rSurrogateInnerLoopGeometries,
+        IntegrationPointsArrayType& rIntegrationPoints,
+        IntegrationInfo& rIntegrationInfo);
+
+    /**
+     * @brief Create a Brep Volume Sbm Integration Points object
+     * 
+     * @param rIntegrationPoints 
+     * @param rSpansU 
+     * @param rSpansV 
+     * @param rSpansW 
+     * @param rOuterLoops 
+     * @param rInnerLoops 
+     * @param rIntegrationInfo 
+     */
+    static void CreateBrepVolumeSbmIntegrationPoints(
+        IntegrationPointsArrayType& rIntegrationPoints,
+        const std::vector<double>& rSpansU,
+        const std::vector<double>& rSpansV,
+        const std::vector<double>& rSpansW,
+        GeometrySurrogateArrayType& rOuterLoops,
+        GeometrySurrogateArrayType& rInnerLoops,
+        IntegrationInfo& rIntegrationInfo);
+
+private:
+    
+    /**
+     * @brief 
+     * 
+     * @param rSpans 
+     * @param coord 
+     * @return int 
+     */
+    static int FindKnotSpans1D(
+        const std::vector<double>& rSpans, const double coord);
```

Comment:
```suggestion
    static int FindKnotSpans1D(
        const std::vector<double>& rSpans,
        const double coord);
```
Besides, why do we need it to be `static`?

Reply:
It's static because it doesn't use any class members. This way, we can call it without creating an object, no?

Reply:
Yes, but you're making it `private` so effectively you cannot call it from elsewhere. Anyhow, leave it like this if you want and if at some point we need to call it from outside we can just move it to the `public` section.